test/unit/org/apache/cassandra/io/sstable/SSTableReaderTest.java

```diff
@@ -915,6 +932,111 @@ public void testVerifyCompressionInfoExistencePasses()
         SSTableReader.verifyCompressionInfoExistenceIfApplicable(desc, components);
     }
 
+    @Test
+    public void testBloomFilterIsCreatedOnLoad() throws IOException
+    {
+        BloomFilter.recreateOnFPChanceChange = true;
+
+        final int numKeys = 100;
+        final Keyspace keyspace = Keyspace.open(KEYSPACE1);
+        final ColumnFamilyStore cfs = keyspace.getColumnFamilyStore(CF_STANDARDNOBLOOMFILTER);
+
+        SSTableReader sstable = getNewSSTable(cfs, numKeys, 1);
+        Assert.assertTrue(sstable.getBloomFilterSerializedSize() == 0);
+        Assert.assertSame(FilterFactory.AlwaysPresent, sstable.getBloomFilter());
+
+        // should do nothing
+        checkSSTableOpenedWithGivenFPChance(sstable, 1, false, numKeys);
+
+        // should create BF because the FP has changed
+        checkSSTableOpenedWithGivenFPChance(sstable, BloomCalculations.minSupportedBloomFilterFpChance(), true, numKeys);
+        checkSSTableOpenedWithGivenFPChance(sstable, 0.05, true, numKeys);
+        checkSSTableOpenedWithGivenFPChance(sstable, 0.1, true, numKeys);
+
+        // should deserialize the existing BF
+        checkSSTableOpenedWithGivenFPChance(sstable, 0.1, true, numKeys);
```

Comment:
It seems to be, but first one is expected to recreate the filter because of the fp chance change while the other one is expected to just load since fp chance is not changed since the last attempt. I will add explicit assertions about whether the file was created or not